lib/networks.js

```diff
@@ -118,134 +118,96 @@ function addNetwork(data) {
+            networks.splice(i, 1);
+        }
     }
-  }
-  for (var key in networkMaps) {
-    if (networkMaps[key] === network) {
-      delete networkMaps[key];
+    for (var key in networkMaps) {
+        if (networkMaps[key] === network) {
+            delete networkMaps[key];
+        }
     }
-  }
 }
 
 addNetwork({
-  name: 'livenet',
-  alias: 'mainnet',
-  pubkeyhash: 0x2089,
-  privatekey: 0x80,
-  scripthash: 0x2096,
-  xpubkey: 0x0488b21e,
-  xprivkey: 0x0488ade4,
-  zaddr: 0x169a,
-  zkey: 0xab36,
-  networkMagic: 0x63617368,
-  port: 9033,
-  dnsSeeds: [
-    'dnseed.zensystem.io',
-    'dnsseed.zenseed.network',
-    'zpool.blockoperations.com',
-    'node1.zenchain.info',
-    'mainnet.zensystem.io'
-  ]
+    name: 'livenet',
+    alias: 'mainnet',
+    pubkeyhash: 0x2089,
+    privatekey: 0x80,
+    scripthash: 0x2096,
+    xpubkey: 0x0488b21e,
+    xprivkey: 0x0488ade4,
+    zaddr: 0x169a,
+    zkey: 0xab36,
+    networkMagic: 0x63617368,
+    port: 9033,
+    dnsSeeds: [
+        'dnseed.zensystem.io',
+        'dnsseed.zenseed.network',
+        'zpool.blockoperations.com',
+        'node1.zenchain.info',
+        'mainnet.zensystem.io'
+    ]
 });
 
-/**
- * @instance
- * @member Networks#livenet
- */
-var livenet = get('livenet');
+var zen_mainnet = get('livenet');
 
 addNetwork({
-  name: 'testnet',
-  alias: 'regtest',
-  pubkeyhash: 0x2098,
-  privatekey: 0xef,
-  scripthash: 0x2092,
-  xpubkey: 0x043587cf,
-  xprivkey: 0x04358394,
-  zaddr: 0x16b6,
-  zkey: 0xac08,
+    name: 'testnet',
+    alias: 'regtest',
+    pubkeyhash: 0x2098,
+    privatekey: 0xef,
+    scripthash: 0x2092,
+    bech32prefix: 'bcrt',
+    xpubkey: 0x043587cf,
+    xprivkey: 0x04358394,
+    networkMagic: 0xfa1af9bf,
+    port: 18233,
+    zaddr: 0x16b6,
+    zkey: 0xac08,
+    dnsSeeds: [
+        'dnsseed.testnet.z.cash'
+    ]
 });
 
 /**
  * @instance
  * @member Networks#testnet
  */
-var testnet = get('testnet');
+var zen_testnet = get('testnet');
 
-// Add configurable values for testnet/regtest
-
-var TESTNET = {
-  PORT: 18233,
-  NETWORK_MAGIC: BufferUtil.integerAsBuffer(0xfa1af9bf),
-  DNS_SEEDS: [
-    'dnsseed.testnet.z.cash',
-  ]
-};
-
-for (var key in TESTNET) {
-  if (!_.isObject(TESTNET[key])) {
-    networkMaps[TESTNET[key]] = testnet;
-  }
-}
-
-var REGTEST = {
-  PORT: 18444,
-  NETWORK_MAGIC: BufferUtil.integerAsBuffer(0xaae83f5f),
-  DNS_SEEDS: []
-};
-
-for (var key in REGTEST) {
-  if (!_.isObject(REGTEST[key])) {
-    networkMaps[REGTEST[key]] = testnet;
-  }
-}
-
-Object.defineProperty(testnet, 'port', {
-  enumerable: true,
-  configurable: false,
-  get: function() {
-    if (this.regtestEnabled) {
-      return REGTEST.PORT;
-    } else {
-      return TESTNET.PORT;
-    }
-  }
+addNetwork({
+    name: 'regtest',
+    alias: 'dev',
+    pubkeyhash: 0x2098,
+    privatekey: 0xef,
+    scripthash: 0x2092,
+    bech32prefix: 'bcrt',
+    xpubkey: 0x043587cf,
+    xprivkey: 0x04358394,
+    networkMagic: 0x2f54cc9d,
```

Comment:
Just curious, what is networkMagic?

Reply:
NetworkMagic are bits that are used as a way to identify the separate messages sent between nodes on the network.